src/test/java/uk/gov/hmcts/cmc/claimstore/events/DocumentUploadHandlerTest.java

```diff
@@ -5,59 +5,201 @@
+    private SettlementAgreementCopyService settlementAgreementCopyService;
+    @Mock
+    private SealedClaimPdfService sealedClaimPdfService;
+    @Mock
+    private ClaimIssueReceiptService claimIssueReceiptService;
+    @Mock
+    private DefendantPinLetterPdfService defendantPinLetterPdfService;
+    @Mock
+    private DocumentsService documentService;
+
+    private DocumentUploadHandler documentUploadHandler;
+
+    private final ArgumentCaptor<PDF> argumentCaptor = ArgumentCaptor.forClass(PDF.class);
+    private final DefendantResponseEvent defendantResponseEvent = new DefendantResponseEvent(
         SampleClaimIssuedEvent.CLAIM_WITH_RESPONSE,
         AUTHORISATION
     );
-
-    private static final DefendantResponseEvent RESPONSE_EVENT_WITHOUT_RESPONSE = new DefendantResponseEvent(
+    private final DefendantResponseEvent defendantResponseEventWithoutResponse = new DefendantResponseEvent(
         SampleClaimIssuedEvent.CLAIM_NO_RESPONSE,
         AUTHORISATION
     );
-    private DocumentUploadHandler documentUploadHandler;
+    private final CountyCourtJudgmentEvent ccjWithoutAdmission = new CountyCourtJudgmentEvent(
+        SampleClaimIssuedEvent.CLAIM_WITH_DEFAULT_CCJ,
+        AUTHORISATION
+    );
+    private final AgreementCountersignedEvent offerMadeByClaimant =
+        new AgreementCountersignedEvent(SampleClaim.getDefault(),
+            MadeBy.CLAIMANT,
+            AUTHORISATION);
+    private final CountersignSettlementAgreementEvent countersignSettlementAgreementEvent =
+        new CountersignSettlementAgreementEvent(SampleClaim.builder().withSettlement(mock(Settlement.class)).build(),
+            AUTHORISATION);
 
     @Before
     public void setUp() {
-        documentUploadHandler = new DocumentUploadHandler(publisher,
-                            defendantResponseReceiptService);
+        documentUploadHandler = new DocumentUploadHandler(defendantResponseReceiptService,
+            countyCourtJudgmentPdfService,
+            settlementAgreementCopyService,
+            sealedClaimPdfService,
+            claimIssueReceiptService,
+            defendantPinLetterPdfService,
+            documentService);
+    }
+
+    @Test
+    public void citizenClaimIssuedEventTriggersDocumentUpload() {
+        Claim claim = SampleClaim.getDefault();
+        CitizenClaimIssuedEvent event = new CitizenClaimIssuedEvent(claim, pin, submitterName, AUTHORISATION);
+        documentUploadHandler.uploadDocument(event);
+        verify(documentService, times(3))
+            .uploadToDocumentManagement(argumentCaptor.capture(), anyString(), any(Claim.class));
+        List<PDF> capturedDocuments = argumentCaptor.getAllValues();
+        List<ClaimDocumentType> expectedClaimDocumentTypes = Arrays.asList(SEALED_CLAIM,
+            DEFENDANT_PIN_LETTER,
+            CLAIM_ISSUE_RECEIPT);
+        capturedDocuments.forEach(document ->
+            assertTrue(expectedClaimDocumentTypes.contains(document.getClaimDocumentType())));
```

Comment:
This may assert true even SEALED_CLAIM is called three times. So better to check contains all with expected and actual.